connectors/byoei.py

```diff
@@ -167,25 +177,30 @@ async def run(self):
 class Fetcher:
     """Grabs data and adds them in the queue for the bulker.
 
-    This class runs a coroutine that puts docs in `queue`, given
-    a document generator.
+    This class runs a coroutine that puts docs in `queue`, given a document generator.
+
+    Arguments:
+    - queue: an `asyncio.Queue` to put docs in
+    - index: the target Elasticsearch index
+    - existing_ids: a list of existing ELasticsearch documents ids found in the index
+    - filter: a `Filter` instance -- default: `None`
+    - sync_rules_enabled: if `True`, we apply rules -- default: `False`
```

Comment:
Why is this param is added in the first place? I think it's redundant to `Fetcher`, which should only care about `filter_`.

Reply:
it's used to instantiate `BasicRuleEngine`, which will run for each document with `self.basic_rule_engine.should_ingest(doc)` to decide if we drop it or not